### ISSUE TITLE
Crash upon loading world

### ISSUE THREAD

**Issue by Zsashas** [BluSunrize/ImmersiveEngineering#983]
https://gist.github.com/Zsashas/39df5fb916564a5997ee

**Comment by theunkn0wn1** [comment]
First please remove Immersive Integration, It is out of date.
The log reports you are using II version 0.6.7, this has not been updated to the new IE release

**Comment by XFactHD** [comment]
Try updating java to the latest version of java 8.

**Comment by malte0811** [comment]
@BluSunrize did you compile with java 8? This was reported before, but always with custom buils using java 8.

**Comment by BluSunrize** [comment]
Ah shit, I might have. I switched to J8 recently due to Forge 1.8 requirements. But I was pretty sure that I still compiled on J7, since I didn't change my buildscript for 1.7...

**Comment by Bogdan-G** [comment]
I have the same error with the latest mod at the entrance to the single world:
https://gist.github.com/Bogdan-G/a92261f8aea5984bb884

**Comment by malte0811** [comment]
I have looked at the compiled files, they are definitely trying to call the Java 8 version of `keySet()`.
From the official documentation for `javac` I would assume that the source/target settings (l. 24-25 in build.gradle) only prevent you from using lambda functions and other stuff that didn't exist at all in J7. Since they only changed the signature of the method, `javac` doesn't know that it has to treat `keySet()` differently when trying to be compatible with J7.
`gradlew.bat` uses the default java version, as defined by the `JAVA_HOME` environment variable. I haven't tried this, but to make it use J7, I suggest replacing `JAVA_HOME` with `JAVA7_HOME` in that file and defining that variable either globally or at the beginning of `gradlew.bat` (`set JAVA7_HOME=<the path to your JDK folder (not the bin folder, one layer above that)>`)

**Comment by BluSunrize** [comment]
Hrrm. I'll give that a shot. It's kinda silly that it effectively disregards the Java version specified in the .gradle file...

**Comment by BluSunrize** [comment]
@Zsashas @Bogdan-G please try again with this version:
https://www.dropbox.com/s/wjyrhdkmbalytlu/ImmersiveEngineering-0.7.2_Java7.jar

And report back.
Note that this is a version solely designed for reaffirming that bug. Please do not distribute it.If this fix helps, I will release an official update.

**Comment by Bogdan-G** [comment]
@BluSunrize problem not solved, if add file to modpack -> dont start game (JVM crash)
if only IE mod and crash(the world has gone, and a minute later departure):  https://gist.github.com/Bogdan-G/5939e49a8ba47b0a67ff
PS I some times even overwrite the file, download suddenly awry. It also did not help.

**Comment by Zsashas** [comment]
Downloaded it, added it to my pack, absolutely no difference. Exact same crash.

**Comment by BluSunrize** [comment]
Fuck.
Alright, I have another idea.

**Comment by BluSunrize** [comment]
@Zsashas @Bogdan-G 
Try this one:
https://www.dropbox.com/s/9mu89snbhsr230o/ImmersiveEngineering-0.7.2_Java7_2.jar

**Comment by Bogdan-G** [comment]
if obly the IE does not seem to notice the crash the game. :/

**Comment by BluSunrize** [comment]
...come again?

**Comment by mindforger** [comment]
wait wait wait ... WHAT ? O_o
`1.7.0_95, Azul Systems, Inc.`

can you install an "ordinary" or "common" java instead?

**Comment by BluSunrize** [comment]
Should be entirely unrelated to the crash though.
If the Java7_2 version I put up /also/ crashes, I want a log of that. Because in theory it should be different, since I cast beforehand to avoid that method signature.

**Comment by mindforger** [comment]
you mean that workaround by using the standard Map interface instead of the actual ConcurrentMap reference?

**Comment by BluSunrize** [comment]
Effectively, yes
I did ((Map)concurrentMap).keySet();

That should have the normal method signature and therefore work, right?

**Comment by mindforger** [comment]
yeah i have also found that while looking up the inconsistency ... but this still is a workaround only .. i am suprised that it has been reported twice as bug to openJDK and both times been invalidated
despite the example i've seen beeing very explicit in doing a clean interface reference first and then call keySet you'r way looks explicit enough to me to succeed

**Comment by BluSunrize** [comment]
hey if it works I'm glad! It's one of those "I'd better not touch it lest it breaks again" cases.
Thing is, I didn't mae ktaht ConcurrentHashMap stuffs, and the fact taht Java just changed a damn method signature like that really ticks me off too.
It's called keySet() for gods sake, why does it need to include values too?! >_<

**Comment by malte0811** [comment]
Just did a bit of google search, have you tried this:http://stackoverflow.com/questions/18487406/how-do-i-tell-gradle-to-use-specific-jdk-version ?

**Comment by BluSunrize** [comment]
Well I even chagned JAVA_HOME for JAVA7_HOME, like you suggested. Taht shoudl ahve worked, but it didn't .-.
And I changed my eclipse to use JRE7 rather than 8. And that didn't help either. So now, I'm doing that cast and hoping that @Zsashas and @Bogdan-G get back to me, because sadly, you can't see method signatures in JDGui, so I can't really see for myself if it had effect.

**Comment by Bogdan-G** [comment]
@mindforger `can you install an "ordinary" or "common" java instead?` - sorry, the company Oracle does not provide links to download Java 7 on its official website.

**Comment by mindforger** [comment]
@Bogdan-G please follow the instructions and test again with the version BluSunrize provided to you

if this still fails, you can still try to use **oracle's java 7 from it's official website** ... it is there you just need to follow the "all java versions link at the bottom" and then "where do i get  java 7" on the left
http://java.com/en/download/faq/java_7.xml

**Comment by BluSunrize** [comment]
http://www.oracle.com/technetwork/java/javase/downloads/jre7-downloads-1880261.html

JAva 7 downloads, for every platform. Right there. Took me 3 seconds of googling "Java 7"

**Comment by Bogdan-G** [comment]
mhm, not JDK? only JRE? I saw where they build 90+ but he was not publicly available.

**Comment by mindforger** [comment]
@Bogdan-G i don't know why exactly they want you to register an account but it does not cost you anything, so just create yourself an account and go on, what is the problem?

 For the Java version you are using you should have made registration too! Or did you rip it off somewhere illegally? in this case i won't support you in any way further!

**Comment by BluSunrize** [comment]
Oi fellas. This ain't the place to discuss legalities of software piracy. We're trying to fix a bug here. Get me some decent crashlogs.

**Comment by Bogdan-G** [comment]
@mindforger you need to specify that I am an employee of a commercial company and obtain permission to download with approval Oracle.
@BluSunrize i check in HotSpot JRE.

**Comment by BluSunrize** [comment]
You can still get J7 for free on Oracles official website .-.

**Comment by Bogdan-G** [comment]
@BluSunrize I checked in January 2016 and did not let me download a file, heh.

**Comment by XFactHD** [comment]
Try deleting your browser history.

**Comment by BluSunrize** [comment]
No Crash?`Great. That's one confirmed fix then. Just waiting on @Zsashas now

**Comment by Zsashas** [comment]
Confirmed successful launch and world load.

**Comment by InuYasha86000** [comment]
I am having the same issue with both 0.7.2 java 7 and 0.7.3, it seems to be something involving Cart livery as I have not tested outside of my modpack:http://pastebin.com/ctQqXWhZ 
I wanted to test it because I want to add it to my modpack I created.

**Comment by XFactHD** [comment]
This has nothing to do with what was discussed here, please make a new issue.

**Comment by BluSunrize** [comment]
Fixed with version 0.7.4
